From 628b84b0203bfc22a6f4af176f25bce52f4755b4 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@gaudilabs.com>
Date: Thu, 08 Feb 2018 17:54:46 +0200
Subject: [PATCH] Add armcp package and all its deps

armcpd is the ARM daemon that handle all dali async events
---

diff --git a/Config.in b/Config.in
index 4cc4d7c..e51bc76 100644
--- a/Config.in
+++ b/Config.in
@@ -1 +1,6 @@
+source "$BR2_EXTERNAL_HL_DALI_PATH/package/armcp/Config.in"
+source "$BR2_EXTERNAL_HL_DALI_PATH/package/specs/Config.in"
+source "$BR2_EXTERNAL_HL_DALI_PATH/package/spdlog/Config.in"
+source "$BR2_EXTERNAL_HL_DALI_PATH/package/specs_external/Config.in"
+
 # Below can add to make it appear in the top-level configuration menu
diff --git a/board/habanalabs/dali/post-build.sh b/board/habanalabs/dali/post-build.sh
index 7df7d2d..55831c1 100755
--- a/board/habanalabs/dali/post-build.sh
+++ b/board/habanalabs/dali/post-build.sh
@@ -14,3 +14,10 @@
 		exit 1
 	fi
 popd
+
+# Add armcpd to dali
+if [ -e ${TARGET_DIR}/etc/inittab ]; then
+    grep -qE '^dali::' ${TARGET_DIR}/etc/inittab || \
+	sed -i '/GENERIC_SERIAL/a# Put armcpd on Dali device\
+dali::respawn:/usr/bin/armcpd # ARM CP' ${TARGET_DIR}/etc/inittab
+fi
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index d743d05..e348e25 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -58,3 +58,4 @@
 BR2_TARGET_UBOOT_ENVIMAGE_REDUNDANT=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS=y
 BR2_PACKAGE_HOST_UBOOT_TOOLS_FIT_SUPPORT=y
+BR2_PACKAGE_ARMCP=y
diff --git a/package/armcp/Config.in b/package/armcp/Config.in
new file mode 100644
index 0000000..7fc3453
--- /dev/null
+++ b/package/armcp/Config.in
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_ARMCP
+	bool "ARM CP"
+	select BR2_PACKAGE_SPECS
+	select BR2_PACKAGE_SPDLOG
+	help
+	  armcp daemon is an agent to host application in monitoring Dali.
diff --git a/package/armcp/armcp.mk b/package/armcp/armcp.mk
new file mode 100644
index 0000000..8ebc954
--- /dev/null
+++ b/package/armcp/armcp.mk
@@ -0,0 +1,13 @@
+ARMCP_VERSION = 1.0.0
+ARMCP_SITE = armcp
+ARMCP_LICENSE = unclear
+
+ARMCP_DEPENDENCIES = specs spdlog
+
+ARMCP_CONF_OPTS = -DTESTS_ENABLED=OFF
+
+define ARMCP_INSTALL_TARGET_CMDS
+        $(INSTALL) -D -m 0755 $(@D)/bin/armcpd $(TARGET_DIR)/usr/bin/armcpd
+endef
+
+$(eval $(cmake-package))
diff --git a/package/spdlog/Config.in b/package/spdlog/Config.in
new file mode 100644
index 0000000..b288a86
--- /dev/null
+++ b/package/spdlog/Config.in
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_SPDLOG
+	bool "SPDLOG"
+	select BR2_PACKAGE_SPECS_EXTERNAL
+	help
+	  Super fast C++ Logging Library
diff --git a/package/spdlog/spdlog.mk b/package/spdlog/spdlog.mk
new file mode 100644
index 0000000..aea85b0
--- /dev/null
+++ b/package/spdlog/spdlog.mk
@@ -0,0 +1,7 @@
+SPDLOG_VERSION = 1.0.0
+SPDLOG_SITE = spdlog
+SPDLOG_LICENSE = unclear
+
+SPDLOG_DEPENDENCIES = specs_external
+
+$(eval $(generic-package))
diff --git a/package/specs/Config.in b/package/specs/Config.in
new file mode 100644
index 0000000..5f417d4
--- /dev/null
+++ b/package/specs/Config.in
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_SPECS
+	bool "SPECS"
+	help
+	  Dali specifications used by armcp.
diff --git a/package/specs/specs.mk b/package/specs/specs.mk
new file mode 100644
index 0000000..e4b7968
--- /dev/null
+++ b/package/specs/specs.mk
@@ -0,0 +1,5 @@
+SPECS_VERSION = 1.0.0
+SPECS_SITE = specs
+SPECS_LICENSE = unclear
+
+$(eval $(generic-package))
diff --git a/package/specs_external/Config.in b/package/specs_external/Config.in
new file mode 100644
index 0000000..b73bc60
--- /dev/null
+++ b/package/specs_external/Config.in
@@ -0,0 +1,4 @@
+config BR2_PACKAGE_SPECS_EXTERNAL
+	bool "SPECS_EXTERNAL"
+	help
+	  Dali external specifications used by spdlog.
diff --git a/package/specs_external/specs_external.mk b/package/specs_external/specs_external.mk
new file mode 100644
index 0000000..ff43e27
--- /dev/null
+++ b/package/specs_external/specs_external.mk
@@ -0,0 +1,5 @@
+SPECS_EXTERNAL_VERSION = 1.0.0
+SPECS_EXTERNAL_SITE = specs_external
+SPECS_EXTERNAL_LICENSE = unclear
+
+$(eval $(generic-package))
